vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -655,6 +687,116 @@ void vtkF3DRenderer::ConfigureGridUsingCurrentActors()
+    {
+      size_t nonzerosI = 0;
+      size_t nonzerosJ = 0;
+      for (size_t j = 0; j < 3; ++j)
+      {
+        if (::abs(m->Element[i][j]) > tol)
+        {
+          nonzerosI++;
+        }
+        if (::abs(m->Element[j][i]) > tol)
+        {
+          nonzerosJ++;
+        }
+      }
+      if (nonzerosI > 1 || nonzerosJ > 1)
+      {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  /* Use `PokeMatrix` around the call to `GetBounds()` to extend box.
+   * Only gives the thightest bounds if the transformation is axis-aligned. */
+  const auto extendBoxAxisAligned = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkNew<vtkMatrix4x4> tmpMatrix;
+    vtkMatrix4x4::Multiply4x4(matrix, prop3d->GetMatrix(), tmpMatrix);
+    prop3d->PokeMatrix(tmpMatrix);
+
+    box.AddBounds(prop3d->GetBounds());
+
+    prop3d->PokeMatrix(nullptr);
+  };
+
+  /* Use custom logic to extend box.
+   * Should give the tightest bounds even when non-axis-aligned */
+  const auto extendBoxArbitrary = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkActor* actor = vtkActor::SafeDownCast(prop3d);
+    if (actor)
+    {
+      vtkPolyDataMapper* polyMapper = vtkPolyDataMapper::SafeDownCast(actor->GetMapper());
+      if (polyMapper)
+      {
+        vtkNew<vtkMatrix4x4> tmpMatrix;
+        vtkMatrix4x4::Multiply4x4(matrix, actor->GetMatrix(), tmpMatrix);
+        vtkPolyData* polydata = polyMapper->GetInput();
+        if (polydata)
+        {
+          double p[4] = { 0, 0, 0, 1 };
+          double q[4];
+          for (vtkIdType i = 0; i < polydata->GetNumberOfPoints(); ++i)
+          {
+            polydata->GetPoint(i, p);
+            tmpMatrix->MultiplyPoint(p, q);
+            box.AddPoint(q);
+          }
+          return true;
+        }
+      }
+    }
+    return false;
+  };
+
+  const bool isAxisAligned = isMatrixAxisAligned(matrix);
+  vtkBoundingBox box;
+
+  /* use `ComputeVisiblePropBounds()`'s logic to iterate `vtkProp3D`s contributing to the bounds */
+  vtkProp* prop;
+  vtkCollectionSimpleIterator pit;
+  for (this->Props->InitTraversal(pit); (prop = this->Props->GetNextProp(pit));)
+  {
+    if (prop->GetVisibility() && prop->GetUseBounds())
+    {
+      const double* bounds = prop->GetBounds();
+      if (bounds != nullptr && vtkMath::AreBoundsInitialized(bounds))
+      {
+        vtkProp3D* prop3d = vtkProp3D::SafeDownCast(prop);
+        if (prop3d)
+        {
+          if (isAxisAligned)
+          {
+            extendBoxAxisAligned(prop3d, box);
+          }
+          else
+          {
+            if (!extendBoxArbitrary(prop3d, box))
+            {
+              const std::string classname = std::string(prop3d->GetClassName());
+              F3DLog::Print(F3DLog::Severity::Warning,
+                "Could not properly account for " + classname +
+                  " in non-axis-aligned bounds computation");
+              extendBoxAxisAligned(prop3d, box);
```

Comment:
I believe this block is needed as it _could_ be reach if we had some unanticipated type of actors. Obviously we'd want to anticipate as much as possible which makes pretty hard to cover with a test.
But at the same time it feels wrong to drop the block altogether because such unanticipated actors could come from plugins.

Reply:
what do you mean by unanticipated type of actors ?

Reply:
ha got it, just put an `assert(actor)` in extendBoxArbitrary and return void.

Reply:
I removed the fallback code as discussed on the assumption that we only deal with `vtkPolyDataMapper`s